src/index.ts

```diff
@@ -25,4 +25,21 @@ export default class DataSource {
 
     return CD.get(url, options, cb);
   }
+
+  getAllRows(params: object) {
```

Comment:
@MansurTsutiev Close! The isolation is important, but, more importantly, the reason we wanted to isolate this is that we want to set the expectation that the return value for this method _will be_ the array of rows.

If we can say upfront that users of this method will expect an array of values, we can change _how_ that data is retrieved without worrying about breaking existing usage, as long as an array of values is returned.

In contrast, the way this method is currently written, a user will still have to navigate the final response by going through `response._meta.all_rows`, which defeats the purpose of the isolation and we don't get any benefits. For example, if we decide to move `all_rows` under a different path e.g. `rows`, every user of this library would need to change their reference from `response._meta.all_rows` to `response._meta.rows`.

If we instead do that for the user and navigate to `all_rows` and return the value stored there, they don't need to worry about that and they know that whenever they call `getAllRows`, it's always an array.
```suggestion
  getAllRows(params: object) {
    // we can even add on a response type on the method signature above
    return this.getRawData(params).then(function(response) {
      return response._meta.all_rows;
    });
  }
```